### PR TITLE
add isDuplicate helper for create response

### PR DESCRIPTION
**Pull request by sualko**
I think we should add helper functions for all message keys, or at least add the key as const to the corresponding class.

### CODE REVIEW

**Comment by sualko** [comment]
I think it make sense to merge this after #49. The following keys are missing:

create: idNotUnique
join: successfullyJoined, InvalidSession, BigBlueButtonServerError, guestDeny
end: sentEndMeetingRequest
putRecordingTextTrackAnchor: upload_text_track_success, upload_text_track_failed, empty_uploaded_text_track, paramError
deleteRecording: InvalidRecordingId
getRecordings: noRecordings

It's not clear if this list is complete, since BBB did not document them and the are not using one general const file, map or similar :see_no_evil:

File: src/Responses/CreateMeetingResponse.php [hunk 1]
```diff
@@ -123,4 +126,12 @@ public function hasBeenForciblyEnded()
     {
         return $this->rawXml->hasBeenForciblyEnded->__toString() === 'true';
     }
+
+    /**
+     * @return bool
+     */
+    public function isDuplicate()
```
Comment (SamuelWei) [inline src/Responses/CreateMeetingResponse.php:133]: what about idNotUnique?

File: src/Responses/DeleteRecordingsResponse.php [hunk 2]
```diff
@@ -24,6 +24,8 @@
  */
 class DeleteRecordingsResponse extends BaseResponse
 {
+    const KEY_INVALID_ID = 'InvalidRecordingId';
```
Comment (SamuelWei) [inline src/Responses/DeleteRecordingsResponse.php:27]: what about a method like isDuplicate() ?

File: src/Responses/EndMeetingResponse.php [hunk 3]
```diff
@@ -24,6 +24,8 @@
  */
 class EndMeetingResponse extends BaseResponse
 {
+    const KEY_SET_END_MEETING_REQUEST = 'sentEndMeetingRequest';
```
Comment (SamuelWei) [inline src/Responses/EndMeetingResponse.php:27]: what about a method like isDuplicate() ?

File: src/Responses/GetRecordingsResponse.php [hunk 4]
```diff
@@ -26,6 +26,8 @@
  */
 class GetRecordingsResponse extends BaseResponse
 {
+    const KEY_NO_RECORDINGS = 'noRecordings';
```
Comment (SamuelWei) [inline src/Responses/GetRecordingsResponse.php:29]: what about a method like isDuplicate() ?

File: src/Responses/JoinMeetingResponse.php [hunk 5]
```diff
@@ -24,6 +24,11 @@
  */
 class JoinMeetingResponse extends BaseResponse
 {
+    const KEY_SUCCESSFULLY_JOINED = 'successfullyJoined';
```
Comment (SamuelWei) [inline src/Responses/JoinMeetingResponse.php:27]: what about a methods like isDuplicate() ?

File: src/Responses/JoinMeetingResponse.php [hunk 6]
```diff
@@ -24,6 +24,11 @@
...
+    const KEY_INVALID_SESSION     = 'InvalidSession';
+    const KEY_SERVER_ERROR        = 'BigBlueButtonServerError';
+    const KEY_GUEST_DENY          = ' guestDeny';
```
Comment (SamuelWei) [inline src/Responses/JoinMeetingResponse.php:30]: is there a space too much?

File: src/Responses/PutRecordingTextTrackResponse.php [hunk 7]
```diff
@@ -24,6 +24,11 @@
  */
 class PutRecordingTextTrackResponse extends BaseResponseAsJson
 {
+    const KEY_SUCCESS     = 'upload_text_track_success';
```
Comment (SamuelWei) [inline src/Responses/PutRecordingTextTrackResponse.php:27]: what about a methods like isDuplicate() ?

File: src/Responses/PutRecordingTextTrackResponse.php [hunk 8]
```diff
@@ -24,6 +24,11 @@
...
+    const KEY_FAILED      = 'upload_text_track_failed';
+    const KEY_EMPTY       = 'empty_uploaded_text_track';
+    const KEY_PARAM_ERROR = 'paramError';
```
Comment (SamuelWei) [inline src/Responses/PutRecordingTextTrackResponse.php:30]: I think a global list of messageKeys in BBB would have prevented this naming scheme chaos

**Review by SamuelWei — CHANGES_REQUESTED** [review]

**Comment by sualko** [comment]
I did not add those helper functions, because I'm not sure if this is really helping anyone. That's something we should be discuss :smile:

File: src/Responses/JoinMeetingResponse.php (hunk 6 above)
Comment (SamuelWei) [reply src/Responses/JoinMeetingResponse.php:30]: @sualko ?

**Comment by SamuelWei** [comment]
> I did not add those helper functions, because I'm not sure if this is really helping anyone. That's something we should be discuss 😄

I guess you are right, most people would not use it. The only question would be should we add it to have a more equal api behavior or not.

File: src/Responses/JoinMeetingResponse.php (hunk 6 above)
Comment (sualko) [reply src/Responses/JoinMeetingResponse.php:30]: I think you are right.

**Comment by SamuelWei** [comment]
> > The only question would be should we add it to have a more equal api behavior or not.
> 
> We could also remove the one I added 😄

If you would remove it, what would be the sence of this PR?

**Comment by sualko** [comment]
> If you would remove it, what would be the sence of this PR?

We would still have all those keys, which are very important from my pov.

**Comment by SamuelWei** [comment]
> > If you would remove it, what would be the sence of this PR?
> 
> We would still have all those keys, which are very important from my pov.

I added some helpers, please let me know if you like the naming

**Comment by sualko** [comment]
Great work. Looks good.

**Review by SamuelWei — APPROVED** [review]

**Pull request merged**